purchase_order_type/models/purchase_order.py

```diff
@@ -36,19 +20,10 @@ def _default_order_type(self):
                                  ondelete='restrict',
                                  default=_default_order_type)
 
-    @api.multi
-    def onchange_partner_id(self, partner_id):
-        res = super(PurchaseOrder, self).onchange_partner_id(partner_id)
-        if partner_id:
-            partner = self.env['res.partner'].browse(partner_id)
-            if partner.purchase_type:
-                res['value'] = res.get('value', {})
-                res['value'].update({
-                    'order_type': partner.purchase_type.id,
-                })
-        return res
-
-    @api.one
-    @api.onchange('order_type')
-    def onchange_order_type(self):
-        self.invoice_method = self.order_type.invoice_method
+    @api.onchange('partner_id', 'company_id')
+    def onchange_partner_id(self):
```

Comment:
If you don't want to call super as this is an isolate operation, you can do:
```
    @api.onchange('partner_id', 'company_id')
    def onchange_partner_id_purchase_order_type(self):
        if self.partner_id.purchase_type:
            self.order_type = self.partner_id.purchase_type.id
```
Although you don't do this change, change the `if` part to the new API possibilities (no need to check the existence of each field before accessing the subfields).